eZ/Publish/Core/Repository/UserService.php

```diff
@@ -731,42 +690,55 @@ public function updateUser(APIUser $user, UserUpdateStruct $userUpdateStruct): A
             throw new UnauthorizedException('user', 'password');
         }
 
-        $this->repository->beginTransaction();
-        try {
-            $publishedContent = $loadedUser;
-            if ($userUpdateStruct->contentUpdateStruct !== null) {
-                $contentDraft = $contentService->createContentDraft($loadedUser->getVersionInfo()->getContentInfo());
-                $contentDraft = $contentService->updateContent(
-                    $contentDraft->getVersionInfo(),
-                    $userUpdateStruct->contentUpdateStruct
-                );
-                $publishedContent = $contentService->publishVersion($contentDraft->getVersionInfo());
-            }
+        $this->executeUserUpdate($loadedUser, $userUpdateStruct);
 
-            if ($userUpdateStruct->contentMetadataUpdateStruct !== null) {
-                $contentService->updateContentMetadata(
-                    $publishedContent->getVersionInfo()->getContentInfo(),
-                    $userUpdateStruct->contentMetadataUpdateStruct
-                );
-            }
+        return $this->loadUser($loadedUser->id);
+    }
 
-            // User\Handler::update call is currently used to clear cache only
-            $this->userHandler->update(
-                new SPIUser(
-                    [
-                        'id' => $loadedUser->id,
-                        'login' => $loadedUser->login,
-                        'email' => $userUpdateStruct->email ?: $loadedUser->email,
-                    ]
-                )
-            );
+    public function updateUserPassword(APIUser $user, UserUpdateStruct $userUpdateStruct): APIUser
```

Comment:
IMHO is worth to introduce User Password Update Struct and not give false impression that it's possible to do more with this method.

Reply:
@adamwojs The reason a new structure was not introduced is because I would consider **not** dispatching `BeforeUserUpdateEvent` and `UserUpdateEvent` to be a BC break, and they rely on `UpdateUserStruct`. Because of this I was also hesitant to dispatch a new event for it.

Which approach is prefered? Should we still force the dispatch of the user update events? Should new `UserPasswordUpdateStruct` be somehow updated from `UserUpdateStruct` if it changes in the event handler?

Reply:
IMHO you can provide BC in `\eZ\Publish\Core\Event\UserService::updateUserPassword` method. Create UpdateUserStruct from UserUpdatePasswordStruct and dispatch accordingly BeforeUpdateUserEvent/UpdateUserEvent. It's win-win scenario: we will keep BC promise and `updateUserPassword` method signature will be correct. WDYT @alongosz ?

Reply:
`\eZ\Publish\Core\Event\UserService::updateUserPassword` now emits both `UserUpdateEvent` and `UserPasswordUpdateEvent` (as well as `Before...` variants). `UserUpdateEvent` is marked as deprecated inside the method and I think I'll open a separate PR to remove it in 4.0, so we do not forget about it @adamwojs?